go/example_code/s3/delete-buckets-by-prefix/s3_delete_buckets_test.go

```diff
@@ -0,0 +1,146 @@
+   CONDITIONS OF ANY KIND, either express or implied. See the License for the
+   specific language governing permissions and limitations under the License.
+*/
+package main
+
+import (
+    "fmt"
+    "os"
+    "strconv"
+    "strings"
+    "testing"
+    "time"
+
+    guuid "github.com/google/uuid"
+
+    "github.com/aws/aws-sdk-go/aws"
+    "github.com/aws/aws-sdk-go/aws/session"
+    "github.com/aws/aws-sdk-go/service/s3"
+    "github.com/aws/aws-sdk-go/service/s3/s3manager"
+)
+
+func createBucket(sess *session.Session, bucketName string) error {
+    // Create Amazon S3 service client
+    svc := s3.New(sess)
+
+    // Create the S3 Bucket
+    _, err := svc.CreateBucket(&s3.CreateBucketInput{
+        Bucket: aws.String(bucketName),
+    })
+    if err != nil {
+        return err
+    }
+
+    // Wait until bucket is created before finishing
+    err = svc.WaitUntilBucketExists(&s3.HeadBucketInput{
+        Bucket: aws.String(bucketName),
+    })
+    if err != nil {
+        return err
+    }
+
+    return nil
+}
+
+func populateBucket(sess *session.Session, bucket string) error {
+    s3Svc := s3.New(sess)
+
+    // Create an uploader with S3 client and default options
+    uploader := s3manager.NewUploaderWithClient(s3Svc)
+
+    // Add some dummy data to the bucket
+    text := "abc"
+    body := strings.NewReader(text)
+
+    upParams := &s3manager.UploadInput{
+        Bucket: &bucket,
+        Key:    aws.String("dummy"),
+        Body:   body,
+    }
+
+    _, err := uploader.Upload(upParams)
+
+    return err
+}
+
+// listBuckets lists the S3 buckets with the given prefix
+func listBuckets(sess *session.Session, prefix string) error {
+    // Create S3 service client
+    svc := s3.New(sess)
+
+    result, err := svc.ListBuckets(nil)
+    if err != nil {
+        fmt.Println("Could not list buckets")
+        return err
+    }
+
+    for _, b := range result.Buckets {
+        if strings.HasPrefix(*b.Name, prefix) {
+            fmt.Println(*b.Name)
+        }
+    }
+
+    return nil
+}
+
+func TestAudioToText(t *testing.T) {
+    // When the test started
+    thisTime := time.Now()
+    nowString := thisTime.Format("20060102150405")
+    fmt.Println("Starting unit test at " + nowString)
+    fmt.Println("")
+
+    // Initialize a session that the SDK uses to load
+    // credentials from the shared credentials file (~/.aws/credentials)
+    sess := session.Must(session.NewSessionWithOptions(session.Options{
+        SharedConfigState: session.SharedConfigEnable,
+    }))
+
+    // Create a unique GUID
+    id := guuid.New()
+    defaultName := id.String()
+
+    // Create 3 dummy buckets with prefix "dummy-"
+    prefix := "dummy-"
+
+    for i := 0; i < 3; i++ {
+        name := prefix + defaultName + strconv.Itoa(i)
+        err := createBucket(sess, name)
+        if err != nil {
+            fmt.Println("Could not create bucket " + name)
+            os.Exit(1)
```

Comment:
I think Go now has an opinion on using os.Exit() inside a test. (probably not to use it)

Reply:
also, since you are planing on terminating program flow here, the else is unnecessary

Reply:
I'll just return, thanks.

Reply:
since this is a test, and it failed, I think better than returning you could use t methods (t.Errorf or t.Fatal)